### PR TITLE
Fix size function tests on MN

### PR DESCRIPTION
**Pull request by fabriziomello**
In b7e04f17 we removed the useless table lock on hypertable size functions and without this explicit lock the SELECT grant is not necessary anymore and doing it break some MN regression tests.

Also removed the tests for PG16 since MN is not supported.

Disable-check: force-changelog-file

### CODE REVIEW

**Review by akuzm — APPROVED** [review]

File: tsl/test/sql/dist_util.sql.in [hunk 1]
```diff
@@ -313,15 +313,10 @@ SELECT count(*) FROM size_test_table;
...
 SELECT 1 FROM pg_table_size('nondisttable');
-
--- hypertable_size requires SELECT privilege on table
-\set ON_ERROR_STOP 0
 SELECT * FROM hypertable_size('disttable');
 SELECT * FROM hypertable_size('nondisttable');
 SELECT * FROM hypertable_detailed_size('disttable') ORDER BY node_name;
 SELECT * FROM hypertable_detailed_size('nondisttable') ORDER BY node_name;
-\set ON_ERROR_STOP 1
-
```
Comment (mkindahl) [inline tsl/test/sql/dist_util.sql.in:324]: This seems like a weird thing to test at all. It's just an implementation aspect, not a key feature of the function, so why test it at all?

File: tsl/test/expected/dist_util-15.out [hunk 2]
```diff
@@ -972,17 +972,32 @@ SELECT 1 FROM pg_table_size('nondisttable');
...
+
 SELECT * FROM hypertable_detailed_size('disttable') ORDER BY node_name;
-ERROR:  permission denied for table disttable
+ table_bytes | index_bytes | toast_bytes | total_bytes |  node_name  
+-------------+-------------+-------------+-------------+-------------
+       24576 |       49152 |        8192 |       81920 | data_node_1
+       16384 |       32768 |        8192 |       57344 | data_node_2
+           0 |        8192 |           0 |        8192 | 
+(3 rows)
+
```
Comment (mkindahl) [inline tsl/test/expected/dist_util-15.out:994]: With explicit figures like this, isn't it going to change often even for changes that are not related to the function? What are we really testing here? That we can call the function? That we return these specific sizes? Something else?

**Review by mkindahl — APPROVED** [review]
I see no problems with the change, so approving, but it is a strange thing to have a test for at all. What are we really testing here?
